song/song_.py

```diff
@@ -75,13 +75,13 @@ def _edge_curation(self) -> None:
         In this implementation, the graph always keep symmetric.
         """
         i_1 = self.neighbor_idxs[0]
-        for j in self.topology[i_1].keys():
+        for j in list(self.topology[i_1].keys()):
             self.topology[i_1][j] *= self.edge_decay_rate  # Decay edges
             self.topology[j][i_1] *= self.topology[i_1][j]
             if self.topology[i_1][j] < self.min_edge_weight:
                 del self.topology[i_1][j]  # Prune edges
                 del self.topology[j][i_1]
-        for j in self.neighbor_idxs[: self.n_neighbors]:
+        for j in self.neighbor_idxs[1 : self.n_neighbors]:
```

Comment:
self.neighbor_idx[0] is i_1
If this value is not remove, the diagonal components will be included in the topology